### PR TITLE
grpc: redirected request brings proper wallet

### PR DESCRIPTION
**Pull request by noxiouz**

### CODE REVIEW

File: insonmnia/hub/server.go [hunk 1]
```diff
@@ -184,7 +186,40 @@ func (h *Hub) onRequest(ctx context.Context, req interface{}, info *grpc.UnarySe
...
+		return nil, status.Error(codes.Unauthenticated, "no peer info")
+	}
+
+	switch au := p.AuthInfo.(type) {
+	case nil:
+		// GRPC_INSECURE should be set
+		return handler(ctx, req)
+	case util.EthAuthInfo:
+		if au.Wallet != h.ethAddr {
+			if h.acl.Has(au.Wallet) {
```
Comment (3Hren) [inline insonmnia/hub/server.go]: Wait, I'm not sure whether we pass wallet with or without `0x` prefix.

Comment (noxiouz) [reply insonmnia/hub/server.go]: I checked: both of them with 0x. Btw, seems we need special type for such things to hide format related stuff

Comment (Sokel) [reply insonmnia/hub/server.go]: such as go-ethereum/common/Address?

Comment (noxiouz) [reply insonmnia/hub/server.go]: yep, but our sturcts hold it as string

Comment (Sokel) [reply insonmnia/hub/server.go]: typecast to common.Address and return one-formatted string already
see this: https://github.com/sonm-io/core/pull/237/files#diff-3ae177b094178b29d82a6311d08193a1R62

Comment (sshaman1101) [reply insonmnia/hub/server.go]: In case of this, it may be better to move the `compareEthAddr` function to the util package.

**Comment by noxiouz** [comment]
@Sokel @3Hren I changed wallet from string to ethereum.common.Address.

I gonna change all ethAddresses across the repo to this type

**Pull request merged**